src/main/java/com/formulasearchengine/mathmltools/mml/CMMLInfo.java

```diff
@@ -771,6 +741,18 @@ public final CMMLInfo toStrictCmmlCont() {
         return this;
     }
 
+
+    public final String toString(boolean useCache) {
+        if (cachedString == null || !useCache) {
+            synchronized (this) {
```

Comment:
I wanted to save the transformation every time I call "isMatch", but the additional method toString is prone for misuse. I will remove it.